arelle/plugin/validate/EFM/Filing.py

```diff
@@ -159,7 +159,7 @@ def validateFiling(val, modelXbrl, isEFM=False, isGFM=False):
     # efmSubmissionType and efmIxdsType are already set when re-validating after redaction/redline removal
     submissionType = getattr(modelXbrl,'efmSubmissionType', val.params.get("submissionType", ""))
     attachmentDocumentType = getattr(modelXbrl,'efmIxdsType', val.params.get("attachmentDocumentType", "")) # this is different from dei:documentType
```

Comment:
It seems like the intent is that attachmentDocumentType is always a string and should be ensured at assignment?
```suggestion
    attachmentDocumentType = getattr(modelXbrl,'efmIxdsType', val.params.get("attachmentDocumentType")) or "" # this is different from dei:documentType
```

Reply:
Not sure I'm comfortable always making it a string, but for this one case on the pattern match it can't be None.  This will probably evolve a bit because attachmentDocumentType might be inferable from the instance in later processing in Filing.py, so would prefer it remain None when not specified.

Reply:
In that case, we can skip the regex match if we know it won't match.

```python
isFeeTagging = attachmentDocumentType and feeTaggingAttachmentDocumentTypePattern.match(attachmentDocumentType)
```

Reply:
@austinmatherne-wk @hermfischer-wf Hi Guys, Hope all is well. Is it possible to get this fix merged to master today please? I spoke with Herm on the 3rd and he said that the change Austin recommended was fine for now. I see there was a merged commit yesterday but this file was not updated. Thank you as always for your valuable time :)

Reply:
Hi all, 
Out team is also waiting for this fix. This is solving our current issue.

Thanks

Reply:
This was merged and released yesterday ([2.30.5](https://github.com/Arelle/Arelle/releases/tag/2.30.5)).

Reply:
I'm missing something?

this line still the same
https://github.com/Arelle/Arelle/blob/18922e707834b6f30d7ca2e379d7c325fcbcd8db/arelle/plugin/validate/EFM/Filing.py#L161

Reply:
My apologies, I thought you were referring to Herm's change in this PR, not my suggestion. Herm is testing the suggestion now.